sdk/ton/transaction.go

```diff
@@ -0,0 +1,68 @@
+package ton
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/xssnick/tonutils-go/address"
+	"github.com/xssnick/tonutils-go/tvm/cell"
+
+	"github.com/smartcontractkit/mcms/types"
+)
+
+func ValidateAdditionalFields(additionalFields json.RawMessage) error {
+	fields := AdditionalFields{
+		Value: big.NewInt(0),
+	}
+	if len(additionalFields) != 0 {
+		if err := json.Unmarshal(additionalFields, &fields); err != nil {
+			return fmt.Errorf("failed to unmarshal TON additional fields: %w", err)
+		}
+	}
+
+	return fields.Validate()
+}
+
+type AdditionalFields struct {
+	Value *big.Int `json:"value"`
```

Comment:
Should we also add contract type here ? like atpos and sui did. So when we parse the tx in CLDF analyzer we would be able to fetch the correct contract [type](https://github.com/smartcontractkit/chainlink-deployments-framework/blob/35aa857a3dc74bc7d31ba8abccf4c24e678dfd66/experimental/analyzer/aptos_analyzer.go#L27) for the decoder. Or we should just reuse the [OperationMetadata.ContractType](https://github.com/smartcontractkit/mcms/blob/fbe931361ce243c30f57cffecae61e7244440193/types/operation.go#L7) for TON